test/crd_validation_test.go

```diff
@@ -0,0 +1,88 @@
+package test
+
+import (
+	"fmt"
+	"github.com/RHsyseng/operator-utils/pkg/validation"
+	"github.com/ghodss/yaml"
+	"github.com/openshift/ocs-operator/pkg/apis/ocs/v1alpha1"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestSampleCustomResources(t *testing.T) {
+	root := "./../deploy/crds"
+	crdCrMap := map[string]string{
+		"ocs_v1alpha1_ocsinitialization_crd.yaml": "ocs_v1alpha1_ocsinitialization_cr",
+		"ocs_v1alpha1_storagecluster_crd.yaml": "ocs_v1alpha1_storagecluster_cr",
+	}
+	for crd, prefix := range crdCrMap {
+		validateCustomResources(t, root, crd, prefix)
+	}
+}
+
+func validateCustomResources(t *testing.T, root string, crd string, prefix string) {
+	schema := getSchema(t, fmt.Sprintf("%s/%s", root, crd))
+	assert.NotNil(t, schema)
+	walkFunc := func(path string, info os.FileInfo, err error) error {
+		if strings.HasSuffix(info.Name(), "crd.yaml") {
+			//Ignore CRD
+			return nil
+		}
+		if strings.HasPrefix(info.Name(), prefix) {
+			bytes, err := ioutil.ReadFile(path)
+			assert.NoError(t, err, "Error reading CR yaml from %v", path)
+			var input map[string]interface{}
+			assert.NoError(t, yaml.Unmarshal(bytes, &input))
+			assert.NoError(t, schema.Validate(input), "File %v does not validate against the %s CRD schema", info.Name(), crd)
+		}
+		return nil
+	}
+	err := filepath.Walk(root, walkFunc)
+	assert.NoError(t, err, "Error reading CR yaml files from ", root)
+}
+
+func TestCompleteCRD(t *testing.T) {
+	root := "./../deploy/crds"
+	crdStructMap := map[string]interface{}{
+		"ocs_v1alpha1_ocsinitialization_crd.yaml": &v1alpha1.OCSInitialization{},
+		"ocs_v1alpha1_storagecluster_crd.yaml": &v1alpha1.StorageCluster{},
+	}
+	for crd, obj := range crdStructMap {
+		schema := getSchema(t, fmt.Sprintf("%s/%s", root, crd))
+		missingEntries := schema.GetMissingEntries(obj)
+		nestedOmissions := [...]string{
```

Comment:
What would you say to defining this list as a const near the top of the file?

Reply:
Updated. Additional paths from recent  CRD change merges also added.